test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs

```diff
@@ -1329,6 +1329,186 @@ public void GaugeSerializationMultipleMetricPoints(string instrumentName, long[]
+            histogram.Record(longValue.Value, this.TagList);
+            histogram.Record(0, this.TagList);
+        }
+        else
+        {
+            var histogram = meter.CreateHistogram<double>(instrumentName);
+            histogram.Record(doubleValue.Value, this.TagList);
+            histogram.Record(0, this.TagList);
+        }
+
+        meterProvider.ForceFlush();
+
+        var buffer = new byte[65360];
+
+        var testTransport = new TestTransport();
+
+        ConnectionStringBuilder connectionStringBuilder = new ConnectionStringBuilder($"Account={expectedAccount};Namespace={expectedNamespace}");
+        var otlpProtobufSerializer = new OtlpProtobufSerializer(testTransport, addAccountAndNamespace ? connectionStringBuilder : null, addPrepopulatedDimensions ? prepopulatedMetricDimensions : null);
+
+        otlpProtobufSerializer.SerializeAndSendMetrics(buffer, meterProvider.GetResource(), new Batch<Metric>(exportedItems.ToArray(), exportedItems.Count));
+
+        Assert.Single(testTransport.ExportedItems);
+
+        var request = new OtlpCollector.ExportMetricsServiceRequest();
+
+        request.MergeFrom(testTransport.ExportedItems[0]);
+
+        Assert.Single(request.ResourceMetrics);
+
+        Assert.NotNull(request.ResourceMetrics[0].Resource);
+
+        AssertOtlpAttributes(addAccountAndNamespace ? resourceAttributes.Concat(accountAndNamespace) : resourceAttributes, request.ResourceMetrics[0].Resource.Attributes);
+
+        Assert.Single(request.ResourceMetrics[0].ScopeMetrics);
+
+        var scope = request.ResourceMetrics[0].ScopeMetrics[0];
+
+        Assert.Equal(meter.Name, scope.Scope.Name);
+
+        Assert.Single(request.ResourceMetrics[0].ScopeMetrics[0].Metrics);
+
+        var metric = request.ResourceMetrics[0].ScopeMetrics[0].Metrics[0];
+
+        Assert.Equal(instrumentName, metric.Name);
+
+        Assert.NotNull(metric.ExponentialHistogram);
+
+        Assert.Single(metric.ExponentialHistogram.DataPoints);
+
+        var dataPoint = metric.ExponentialHistogram.DataPoints[0];
+
+        // Assert time
+        var metricPointsEnumerator = exportedItems[0].GetMetricPoints().GetEnumerator();
+        metricPointsEnumerator.MoveNext();
+        var metricPoint = metricPointsEnumerator.Current;
+
+        Assert.Equal((ulong)metricPoint.StartTime.ToUnixTimeNanoseconds(), dataPoint.StartTimeUnixNano);
+        Assert.Equal((ulong)metricPoint.EndTime.ToUnixTimeNanoseconds(), dataPoint.TimeUnixNano);
+
+        Assert.Equal(20, dataPoint.Scale);
+        Assert.Equal(1UL, dataPoint.ZeroCount);
+        if (longValue > 0 || doubleValue > 0)
+        {
+            Assert.Equal(2UL, dataPoint.Count);
+        }
+        else
+        {
+            Assert.Equal(1UL, dataPoint.Count);
+        }
+
+        if (longValue.HasValue)
+        {
+            if (longValue > 0)
+            {
+                Assert.Equal((double)longValue, dataPoint.Max);
```

Comment:
Histograms only support double value anyway so we could skip testing this path.

Reply:
done